packages/amplify-codegen-e2e-tests/src/codegen-tests-base/add-codegen.ts

```diff
@@ -29,3 +35,105 @@ export async function testAddCodegen(config: AmplifyFrontendConfig, projectRoot:
     expect(isNotEmptyDir(path.join(projectRoot, config.graphqlCodegenDir))).toBe(true);
     testValidGraphQLConfig(projectRoot, config);
 }
+
+export type TestAddCodegenUninitializedProps = {
+    config: AmplifyFrontendConfig;
+    projectRoot: string;
+    sdlFilename?: string;
+    expectedFilenames: Array<string>;
+    dropAndRunCodegen?: boolean;
+    dropAndRunCodegenStatements?: boolean;
+    dropAndRunCodegenTypes?: boolean;
+    initialFailureMessage?: string;
+};
+
+const assertTypeFileExists = (projectRoot: string): void => {
+    expect(existsSync(path.join(projectRoot, 'src', 'API.ts'))).toBe(true)
+};
+
+export async function testAddCodegenUninitialized({
+    config,
+    projectRoot,
+    sdlFilename,
+    expectedFilenames,
+    dropAndRunCodegen,
+    dropAndRunCodegenStatements,
+    dropAndRunCodegenTypes,
+    initialFailureMessage,
+}: TestAddCodegenUninitializedProps) {
+    // Setup the non-amplify project with schema and pre-existing files
+    const userSourceCodePath = testSetupBeforeAddCodegen(projectRoot, config);
+
+    // Write SDL Schema
+    if (sdlFilename) {
+        const sdlSchema = readFileSync(path.join(__dirname, '..', '..', 'schemas', 'sdl', sdlFilename), 'utf-8');
+        writeFileSync(path.join(projectRoot, sdlFilename), sdlSchema);
+    }
+
+    // add codegen without init
+    switch (config.frontendType) {
+        case AmplifyFrontend.javascript:
+            await addCodegenNonAmplifyJS(projectRoot, initialFailureMessage);
+            break;
+        case AmplifyFrontend.typescript:
+            await addCodegenNonAmplifyTS(projectRoot, initialFailureMessage);
+            break;
+        default:
+            throw new Error(`Received unexpected frontendType ${config.frontendType}`);
+    }
+
+    // return if we expected the add command to fail
+    if (initialFailureMessage) {
+        return;
+    }
+
+    // pre-existing file should still exist
+    expect(existsSync(userSourceCodePath)).toBe(true);
+    // GraphQL statements are generated
+    const generatedStatementFiles = new Set(readdirSync(path.join(projectRoot, config.graphqlCodegenDir)));
+    console.log(`Comparing written files: ${JSON.stringify(Array.from(generatedStatementFiles))} to expected files: ${JSON.stringify(expectedFilenames)}`);
```

Comment:
Do we need to log both arrays here as they are compared on the next line?

Reply:
Yeah, this'll make debugging easier because the comparison line below fails w/ no indication of what the mismatch was,.